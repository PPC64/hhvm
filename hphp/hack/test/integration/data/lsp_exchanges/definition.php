<?hh  //strict

function a_definition(): int {
  return b_definition();
}

function b_definition(): int {
  return 42;
}
