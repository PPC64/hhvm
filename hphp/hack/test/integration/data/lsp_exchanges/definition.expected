[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": false,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":"
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "renameProvider": false,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": [
      {
        "uri": "${php_file_uri}",
        "range": {
          "start": {
            "line": 6,
            "character": 9
          },
          "end": {
            "line": 6,
            "character": 21
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": [
      {
        "uri": "${php_file_uri}",
        "range": {
          "start": {
            "line": 11,
            "character": 18
          },
          "end": {
            "line": 11,
            "character": 29
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 4,
    "result": [
      {
        "uri": "${php_file_uri}",
        "range": {
          "start": {
            "line": 11,
            "character": 18
          },
          "end": {
            "line": 11,
            "character": 29
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 5,
    "result": [
      {
        "uri": "${php_file_uri}",
        "range": {
          "start": {
            "line": 11,
            "character": 18
          },
          "end": {
            "line": 11,
            "character": 29
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 6,
    "result": [
      {
        "uri": "${php_file_uri}",
        "range": {
          "start": {
            "line": 21,
            "character": 18
          },
          "end": {
            "line": 21,
            "character": 29
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 7,
    "result": [
      {
        "uri": "${php_file_uri}",
        "range": {
          "start": {
            "line": 26,
            "character": 6
          },
          "end": {
            "line": 26,
            "character": 8
          }
        }
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 99,
    "result": null
  }
]
